src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/umbBlockListPropertyEditor.component.js

```diff
@@ -261,10 +298,13 @@
             }
             
             var blockEditorModel = {
+                $parentScope: $scope, // pass in a $parentScope, this maintains the scope inheritance in infinite editing
+                $parentForm: parentForm || vm.propertyForm, // pass in a $parentForm, this maintains the FormController hierarchy with the infinite editing view (if it contains a form)
                 hideContent: blockObject.hideContentInOverlay,
                 openSettings: openSettings === true,
                 liveEditing: liveEditing,
                 title: blockObject.label,
+                index: blockIndex,
```

Comment:
I´m not super happy about this 'index', I think that reflects that block editor has to be editing blocks of a list, but I don't think we should dictate that. :-)
If someone wants to have blocks in a _Dictionary_ or just single properties(like a 'left' and 'right' block) they should be able to do so. If I read and understood correctly we are not going to use this Index currently, and reading the _BlockEditorController_ its not begin used. So I think it would be good to not have this.

Currently, the _editBlock_ method is exposed thought vm.blockEditorApi, and is used by customViews to open the infinite editor, for those developers it's not so nice that they have to parse these many attributes, would be nice to keep it as simple as possible. So I would love to hear your thoughts on the options to simplify this as much as possible. One though could be that the _umbBlockListPropertyEditor_ appends a _edit_() method to the block-objects, which gathers the references, index, etc. required for this.

let me know your thoughts, thanks :-)

Reply:
@nielslyngsoe Actually the index probably no longer needs to exist and be passed everywhere. This was an initial requirement when thinking and prototyping some json path stuff with validation. I actually think it can be removed. The reason i kept it in there was because I figured we might end up using the index for other purposes and more information would be better but as you note that might not be required for other editor types. I know we use index for some other things but IIRC we no longer require or use it within the editBlock/editSettings stuff. Are you able to make tasks for this to be cleaned up?

Reply:
Good, Ive made a task for it in AB.